File: implementations/java-springboot/examples/src/main/java/com/example/accountservice/account/domain/TransactionCategory.java
package com.example.accountservice.account.domain;

/**
 * The fixed taxonomy {@code TransactionAutoCategorizer} classifies a withdrawal's merchantName
 * into. Lives here (not in the application layer) for the same reason {@code TransactionType} does
 * — it's a value the domain read/write model carries.
 */
public enum TransactionCategory {
    FOOD,
    TRANSPORT,
    SHOPPING,
    HOUSING,
    MEDICAL,
    ENTERTAINMENT,
    UTILITIES,
    OTHER
}

File: implementations/kotlin-springboot/examples/src/main/kotlin/com/example/accountservice/account/domain/TransactionCategory.kt
package com.example.accountservice.account.domain

// The fixed taxonomy TransactionAutoCategorizer classifies a withdrawal's merchantName into. Lives
// here (not in the application layer) for the same reason TransactionType does — it's a value the
// domain read/write model carries.
enum class TransactionCategory { FOOD, TRANSPORT, SHOPPING, HOUSING, MEDICAL, ENTERTAINMENT, UTILITIES, OTHER }
